<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>864</width>
    <height>607</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Magnetic Field Experiment</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="PlotWidget" name="plot_widget" native="true"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>864</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QDockWidget" name="stepper_motor_dockwidget">
   <property name="windowTitle">
    <string>Stepper Mottor Controler</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QHBoxLayout" name="connection_layout" stretch="2,1,1,1">
         <item>
          <widget class="QComboBox" name="com_ports_combobox"/>
         </item>
         <item>
          <widget class="QPushButton" name="connect_button">
           <property name="text">
            <string>Connect</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="disconnect_button">
           <property name="text">
            <string>Disconnect</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="refresh_button">
           <property name="text">
            <string>Refresh</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="direction_layout">
         <item>
          <widget class="QLabel" name="direction_label">
           <property name="text">
            <string>Direction</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="direction_combobox">
           <item>
            <property name="text">
             <string>Forward</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Backward</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="current_position_text_label">
           <property name="text">
            <string>Current Position</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="current_position_label">
           <property name="styleSheet">
            <string notr="true">font: 75 24pt &quot;Arial&quot;;</string>
           </property>
           <property name="text">
            <string>TextLabel</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="no_of_steps_label">
           <property name="text">
            <string>No. of Steps</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="no_of_steps_spinbox">
           <property name="maximum">
            <number>100000</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="move_to_label">
           <property name="text">
            <string>Move to:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="move_to_spinbox">
           <property name="minimum">
            <number>-999999999</number>
           </property>
           <property name="maximum">
            <number>999999999</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QPushButton" name="move_button">
           <property name="text">
            <string>Move</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="go_to_button">
           <property name="text">
            <string>Go To</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QPushButton" name="set_zero_button">
           <property name="text">
            <string>Set as Zero</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="stop_button">
           <property name="text">
            <string>Stop</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="magnetic_scan_widget">
   <property name="windowTitle">
    <string>Magnetic Field Experiment</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>4</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout" stretch="2,2,1,4">
     <item>
      <layout class="QHBoxLayout" name="scan_cfg_layout" stretch="1,1,1">
       <item>
        <layout class="QHBoxLayout" name="scan_range_layout" stretch="1,2">
         <item>
          <widget class="QLabel" name="scan_range_label">
           <property name="text">
            <string>Scan Range</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="scan_range_spinbox">
           <property name="suffix">
            <string> steps</string>
           </property>
           <property name="prefix">
            <string/>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>99999999</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="scan_steps_layout" stretch="1,2">
         <item>
          <widget class="QLabel" name="scan_steps_label">
           <property name="text">
            <string>Points</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="scan_steps_spinbox">
           <property name="maximum">
            <number>999999999</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="track_every_layout" stretch="1,1">
         <item>
          <widget class="QLabel" name="track_every_label">
           <property name="text">
            <string>Track every</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="track_every_spinbox"/>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="start_stop_layout">
       <item>
        <widget class="QPushButton" name="start_button">
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="stop_scan_button">
         <property name="styleSheet">
          <string notr="true">  color: rgb(255, 0, 4);
  background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:1, y2:0, 
  stop:0 rgba(255, 0, 0, 255),
  stop:0.166 rgba(255, 255, 0, 255),
  stop:0.333 rgba(0, 255, 0, 255),
  stop:0.5 rgba(0, 255, 255, 255),
  stop:0.666 rgba(0, 0, 255, 255),
  stop:0.833 rgba(255, 0, 255, 255),
  stop:1 rgba(255, 0, 0, 255));</string>
         </property>
         <property name="text">
          <string>Stop</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QLabel" name="filename_label">
       <property name="text">
        <string>FILENAME</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="file_io_layout">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="save_layout">
         <item>
          <widget class="QPushButton" name="save_button">
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../../../../artwork/qudi_icons.qrc">
             <normaloff>:/icons/icons/save.png</normaloff>:/icons/icons/save.png</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="delete_button">
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../../../../artwork/qudi_icons.qrc">
             <normaloff>:/icons/icons/delete.png</normaloff>:/icons/icons/delete.png</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="load_layout">
         <item>
          <widget class="QPushButton" name="load_button">
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../../../../artwork/qudi_icons.qrc">
             <normaloff>:/icons/icons/load.png</normaloff>:/icons/icons/load.png</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="previous_button">
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../../../../artwork/qudi_icons.qrc">
             <normaloff>:/icons/icons/previous.png</normaloff>:/icons/icons/previous.png</iconset>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="next_button">
           <property name="text">
            <string/>
           </property>
           <property name="icon">
            <iconset resource="../../../../artwork/qudi_icons.qrc">
             <normaloff>:/icons/icons/next.png</normaloff>:/icons/icons/next.png</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QWidget</extends>
   <header>pyqtgraph</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../../artwork/qudi_icons.qrc"/>
 </resources>
 <connections/>
</ui>
